src/components/NcCheckboxRadioSwitch/NcCheckboxRadioSwitch.vue

```diff
@@ -593,15 +593,20 @@ export default {
 
 			// If the initial value was a boolean, let's keep it that way
 			if (typeof this.checked === 'boolean') {
-				this.$emit('update:checked', !this.isChecked)
+				this.$emit('update:checked', !this.checked)
```

Comment:
Shouldn't be done the same for TYPE_SWITCH?

Reply:
Switch is supposed to have only `boolean` value, so should work